### PR TITLE
dropparty: add

### PR DESCRIPTION
**Pull request by lordzuku**
working & tested

### CODE REVIEW

**Comment by lordzuku** [comment]
https://gyazo.com/b5f96d93310a5be6ccb9a9d572d3f74e

tracks and displays the users path, allowing you to know when potential items will be appearing

**Comment by SRLJustin** [comment]
Checkstyle failed: see https://github.com/runelite-extended/runelite/wiki/Checkstyle-with-IntelliJ-IDEA

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
+		this.client = client;
+		this.plugin = plugin;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		int tiles = plugin.getShowAmmount();
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: final

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		if (tiles == 0)
+		{
+			return null;
+		}
+		List<WorldPoint> path = plugin.getPlayerPath();
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: final

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+		List<WorldPoint> markedTiles = new ArrayList<>();
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: final

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+		for (int i = 0; i < path.size(); i++)
+		{
+			if (i > plugin.getMAXPATHSIZE() || i > (plugin.getShowAmmount() - 1))
+			{
+				break;
+			}
+			if (path.get(i) != null)
+			{
+				LocalPoint local = LocalPoint.fromWorld(client, path.get(i));
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: final

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+				Polygon tilePoly = null;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: null check the local point, assign poly, null check poly.

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+				if (local != null)
+				{
+					tilePoly = Perspective.getCanvasTileAreaPoly(client, local, 1);
+				}
+
+
+				if (tilePoly != null)
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: caseguards > nested ifs

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+		tags = {"Drop", "Party", "marker", "player"},
+		type = PluginType.UTILITY,
+		enabledByDefault = false
+)
+@Singleton
+@Slf4j
+public class DropPartyPlugin extends Plugin
+{
+	@Inject
+	public DropPartyConfig config;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	List<WorldPoint> playerPath = new ArrayList<>();
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private final

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	String playerName = "";
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	int showAmmount = 0;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	int MAXPATHSIZE = 100;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+	Player runningPlayer;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	Color overlayColor;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: private

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+	{
+		eventbus.subscribe(ConfigChanged.class, this, this::onConfigChanged);
+		eventbus.subscribe(GameTick.class, this, this::onGameTick);
+	}
+
+
+	private void onGameTick(GameTick event)
+	{
+		shuffleList();
+		if (playerName.equalsIgnoreCase("") || playerName == null)
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: null check before ignore case, can playerName even be null?

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+		{
+			return;
+		}
+
+		runningPlayer = null;
+
+		for (Player player : client.getPlayers())
+		{
+			if (player.getName() != null)
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyPlugin.java]: caseguards > nested ifs

File: runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java (hunk 3 above)
Comment (lordzuku) [reply runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: is modified later, so not final

Comment (Ganom) [reply runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: the list is final, that doesnt mean the contents are.

Comment (lordzuku) [reply runelite-client/src/main/java/net/runelite/client/plugins/dropparty/DropPartyOverlay.java]: Til

**Pull request merged**